export/securedrop_export/print/service.py

```diff
@@ -241,29 +241,29 @@ def _is_open_office_file(self, filename):
     def _print_file(self, file_to_print):
         # If the file to print is an (open)office document, we need to call unoconf to
         # convert the file to pdf as printer drivers do not support this format
-        if self._is_open_office_file(file_to_print):
-            logger.info("Converting Office document to pdf")
-            folder = os.path.dirname(file_to_print)
-            converted_filename = file_to_print + ".pdf"
-            converted_path = os.path.join(folder, converted_filename)
-            self.safe_check_call(
-                command=["unoconv", "-o", converted_path, file_to_print],
-                error_status=Status.ERROR_PRINT,
+        try:
+            if self._is_open_office_file(file_to_print):
+                logger.info("Converting Office document to pdf")
+                folder = os.path.dirname(file_to_print)
+                converted_filename = file_to_print + ".pdf"
+                converted_path = os.path.join(folder, converted_filename)
+                subprocess.check_call(["unoconv", "-o", converted_path, file_to_print])
+                file_to_print = converted_path
+
+            logger.info(f"Sending file to printer {self.printer_name}")
+
+            subprocess.check_call(
+                ["xpp", "-P", self.printer_name, file_to_print],
             )
-            file_to_print = converted_path
-
-        logger.info(f"Sending file to printer {self.printer_name}")
+            # This is an addition to ensure that the entire print job is transferred over.
+            # If the job is not fully transferred within the timeout window, the user
+            # will see an error message.
+            self._wait_for_print()
 
-        self.safe_check_call(
-            command=["xpp", "-P", self.printer_name, file_to_print],
-            error_status=Status.ERROR_PRINT,
-        )
-        # This is an addition to ensure that the entire print job is transferred over.
-        # If the job is not fully transferred within the timeout window, the user
-        # will see an error message.
-        self._wait_for_print()
+        except subprocess.CalledProcessError as e:
+            raise ExportException(sdstatus=Status.ERROR_PRINT, sderror=e.output)
 
-    def safe_check_call(self, command: str, error_status: Status, ignore_stderr_startswith=None):
+    def safe_check_output(self, command: str, error_status: Status, ignore_stderr_startswith=None):
```

Comment:
I'm kind of meh about the renaming, I don't think it's really either check_call (captures stdout/stderr instead of passing it through) nor check_output (doesn't return stdout). I might have have picked something like `check_call_and_stderr` to indicate that it calls the function and checks the exit code and presence of stderr ("safe" implies it's a security thing, but it's not insecure to not use it...).

This is not a blocking comment, I'll approve regardless.

Reply:
I can rename, I agree.  I do think the behaviour is like(?) check_output, in that it succeeds if it returns 0 and (something) about the output, but I like your name better